### DISCUSSION TITLE
Pass integer from a watchface to an app

### DISCUSSION THREAD

**Discussion by zdunex25** [zepp-health/zeppos-forum#230] [Developer Feedback]
Hi, from what I was able to find there is:
https://docs.zepp.com/docs/watchface/api/hmFS/SysProSetInt/ - for wachfaces
and:
https://docs.zepp.com/docs/reference/device-app-api/newAPI/storage/localStorage/ - for apps
the problem is that the data saved either way is not mutual, so how to properly transfer it?

@MyoungXUE

**Comment by silver-zepp** [comment]
Hey, @zdunex25. Does your watch run on Zepp OS 1.0? If so, the easiest way would be to use **SysProSetInt** with **SysProGetInt** within both APIs (watch face & device) to transfer a variable -> https://docs.zepp.com/docs/1.0/reference/device-app-api/hmFS/SysProSetInt/

The **localStorage** or the **hmFS File System Module** is permanent storage, which is bound to its app/watch face location. So you won't be able to share a variable using this approach.

**Reply by zdunex25** [reply]
I am running os 2.1 and it throws SysProSetInt is unidentified as soon as I try to use it within app environment.

**Reply by silver-zepp** [reply]
I see. For 2.0+, you want to use the [sessionStorage](https://docs.zepp.com/docs/reference/device-app-api/newAPI/storage/sessionStorage). But I don't think it will be compatible with the old **SysPro** approach. Though you can try and see if that works. Then on Monday @MyoungXUE will hopefully give us more information on this.

**Reply by zdunex25** [reply]
I tried sessionStorage as well, did not work tho, let's wait for @MyoungXUE

**Reply by zdunex25** [reply]
@silver-zepp due to the fact that there is many libraries based on api 1.0 I decided to do the same, so I started from scratch and my app is perfect extension for my watchface now, thanks for pointing that out

**Reply by silver-zepp** [reply]
Awesome @zdunex25. Glad we have sorted this out!
